### ISSUE TITLE
strange result with SDL_Vulkan_GetInstanceExtensions function

### ISSUE THREAD

**Issue by miyoku157** [libsdl-org/SDL#8468]
Hi,
I'm using the current master branch to follow the SDL 3.0 development, and i saw recently an update on the function `SDL_Vulkan_GetInstanceExtensions` which now return directly a char const* const*.

So maybe the problem i have come from the fact i'm using the cpp binding of vulkan (vulkan.hpp)
but when i'm using the function i'm either getting 0 extensions or strange number like  21911.
In this sample projects for instance: https://gist.github.com/miyoku157/f0934e45237e4d68bc57df24598d7afb
BTW: the main.cpp must be inside a src folder
I always got 0 extensions and if i remember well, before i got two (KHR_swapchain and another i think?)

Which make me think about another question with this modification, how should we add the  extensions we need like : "VK_EXT_DEBUG_UTILS_EXTENSION_NAME".

**Comment by slouken** [comment]
@ilyas-taouaou, any ideas?

**Comment by ilyas-taouaou** [comment]
> Hi, I'm using the current master branch to follow the SDL 3.0 development, and i saw recently an update on the function `SDL_Vulkan_GetInstanceExtensions` which now return directly a char const* const*.
> 
> So maybe the problem i have come from the fact i'm using the cpp binding of vulkan (vulkan.hpp) but when i'm using the function i'm either getting 0 extensions or strange number like 21911. In this sample projects for instance: https://gist.github.com/miyoku157/f0934e45237e4d68bc57df24598d7afb BTW: the main.cpp must be inside a src folder I always got 0 extensions and if i remember well, before i got two (KHR_swapchain and another i think?)
> 
> Which make me think about another question with this modification, how should we add the extensions we need like : "VK_EXT_DEBUG_UTILS_EXTENSION_NAME".

Before, SDL was returing the extensions in an allocation, now it just returns a const pointer to a static array.

If you don't need to add additional extensions, you can just pass the extensions pointer directly to vulkan, otherwise you'll have to do what SDL was doing before which is allocating memory and copying the data of the returned extensions, and then add some new elements to your new allocated memory.

I'll send you some example code on how to do it soon.

**Comment by ilyas-taouaou** [comment]
but basically you'll have to copy the SDL returned extensions to a new vector

**Comment by ilyas-taouaou** [comment]
Oh you're also passing it to the wrong field
```
    inst_info.ppEnabledLayerNames = test;
    inst_info.enabledExtensionCount = count;
```

**Comment by icculus** [comment]
> i'm either getting 0 extensions or strange number like 21911

- Make sure you're using the right SDL library; if you're calling into a build from before we made this change, by accident, you'll definitely get nonsense numbers here.
- Make sure it isn't failing! You didn't initialize `count`, and you didn't check if the function failed by returning a NULL pointer, so it's possible nothing ever initialized this value, causing strange numbers.

**Comment by miyoku157** [comment]
> > i'm either getting 0 extensions or strange number like 21911
> 
>     * Make sure you're using the right SDL library; if you're calling into a build from before we made this change, by accident, you'll definitely get nonsense numbers here.
> 
>     * Make sure it isn't failing! You didn't initialize `count`, and you didn't check if the function failed by returning a NULL pointer, so it's possible nothing ever initialized this value, causing strange numbers.

Alright so i tested everything you told. And finally it was because i didn't initialize `count`. I was still following the example here: https://wiki.libsdl.org/SDL3/SDL_Vulkan_GetInstanceExtensions
@icculus if you have a sample of how it should be done, i must say i'm quite interest. 
But for this issue, it's solved

**Comment by icculus** [comment]
> And finally it was because i didn't initialize count.

Did SDL_Vulkan_GetInstanceExtensions return NULL? If it didn't return an error (NULL in this case) but also failed to initialize count, then that's definitely an SDL bug.

**Comment by ilyas-taouaou** [comment]
> > And finally it was because i didn't initialize count.
> 
> Did SDL_Vulkan_GetInstanceExtensions return NULL? If it didn't return an error (NULL in this case) but also failed to initialize count, then that's definitely an SDL bug.

The docs should be updated, cause it will never return an error
It's too simply exposing the internal static array
And if the count is not null, it writes the count to it

**Comment by icculus** [comment]
>  And if the count is not null, it writes the count to it

@miyoku157 Are you using Wayland, by any chance?

```c
char const* const* Wayland_Vulkan_GetInstanceExtensions(SDL_VideoDevice *_this,
                                              Uint32 *count)
{
    static const char *const extensionsForWayland[] = {
        VK_KHR_SURFACE_EXTENSION_NAME, VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME
    };
    return extensionsForWayland;
}
```

...we forgot to set the count variable.

**Comment by icculus** [comment]
> The docs should be updated, cause it will never return an error

Let's leave it as it is, in case it becomes more dynamic later (some future optional extensions show up, and we run out of memory allocating an internal array, or whatever).

**Comment by miyoku157** [comment]
@icculus yes i'm on archlinux using wayland. It's true i forget to say it.

**Comment by icculus** [comment]
> yes i'm on archlinux using wayland

Okay, this was definitely an SDL bug, and I've pushed a fix for it now. Thanks!

**Comment by GloryOfNight** [comment]
Hi, I just same way compiled and using latest main branch and on my windows application I get 0 instance extensions count returned by SDL_Vulkan_GetInstanceExtensions. 

I've added SDL_Vulkan_LoadLibrary(nullptr) call before calling  SDL_Vulkan_GetInstanceExtensions. And still 0 extensions returned. 

In version I've used before latest one, there was an surface extension. Now its noting. Is that intended?
